### PR TITLE
Test special cases

### LINKED ISSUE #2889: Test special cases in user finding
Test the handling of special cases in user finding. Generally they should return errors in the 400 range. See also https://github.com/CSCfi/rems/issues/2856

- [ ] "no user found" and 
- [ ] "multiple users found with same identity value".

### PR DESCRIPTION
**Pull request by Macroz**
Close #2889.

Simple error page for users trying to log in without name or email.

problem

# Checklist for author

Remove items that aren't applicable, check items that are done.

## Reviewability
- [x] Link to issue
- [x] Consider adding screenshots for ease of review

## Backwards compatibility
- [ ] API is backwards compatible or completely new (theoretically breaks as new error codes are returned)

## Documentation
- [x] Update changelog if necessary
- [ ] API is documented and shows up in Swagger UI (the 400 error is undocumented, doesn't seem worthwhile to fix in compojure-api)

## Testing
- [x] Complex logic is unit tested
- [x] Valuable features are integration / browser / acceptance tested automatically

### CODE REVIEW

File: test/clj/rems/auth/test_oidc.clj [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+            [rems.auth.oidc :as oidc]
+            [rems.config]
+            [rems.ga4gh]
+            [rems.jwt]
+            [rems.json :as json]
+            [rems.testing-util :refer [with-fake-login-users]]))
+
+(use-fixtures :once api-fixture)
+
+(deftest test-user-does-not-exist
```
Comment (aatkin) [inline test/clj/rems/auth/test_oidc.clj:14]: these two tests `test-user-does-not-exist` and `test-user-has-no-name` seem to be identical now?

File: src/cljs/rems/spa.cljs [hunk 2]
```diff
@@ -532,6 +558,29 @@
 (secretary/defroute "/forbidden" []
   (rf/dispatch [:set-active-page :forbidden]))
 
+;; XXX: could use schema?
+(defn- fix-error
+  "Fixes the error value transforming strings into keywords etc."
```
[the remark this answers is not in the document]
Comment (Macroz) [reply src/cljs/rems/spa.cljs:563]: The map exploded into query parameters comes in as strings. So we want to transform it to keywords where expected. This is not an API call where we handle it in Transit etc.

File: test/clj/rems/auth/test_oidc.clj (hunk 1 above)
Comment (Macroz) [reply test/clj/rems/auth/test_oidc.clj:14]: There is a lot of setup involved to run them which can be perhaps shared. I'll see what I can do.

File: src/cljs/rems/spa.cljs (hunk 2 above)
[the remark this answers is not in the document]
Comment (Macroz) [reply src/cljs/rems/spa.cljs:563]: Sadly `keywordize-keys` is also not enough as the keywords we need are in the values.

[the remark this answers is not in the document]
Comment (aatkin) [reply src/cljs/rems/spa.cljs:563]: check! could the function name reflect this behaviour a little better? i read `fix` as repairing something broken and we have used `format` and `map` before. something like `map-error-from-query-params`?

[the remark this answers is not in the document]
Comment (Macroz) [reply src/cljs/rems/spa.cljs:563]: We have `fix-row-from-db` and `fix-event-from-db` for this kind of functions already? So I'll rename if `fix-error-from-query-string`.

**Review by aatkin — APPROVED** [review]

**Pull request merged**